samples/diagnostics/program-visualization/TraceExample.ipynb

```diff
@@ -0,0 +1,553 @@
+    "    X(target);\n",
+    "    H(target);\n",
+    "\n",
+    "    within {\n",
+    "        ApplyToEachA(H, queryRegister);\n",
+    "    } apply {\n",
+    "        // |𝑥, 𝑦〉 ↦ |𝑥, 𝑦 ⊕ 𝑓(𝑥)〉.\n",
+    "        Uf(queryRegister, target);\n",
+    "    }\n",
+    "\n",
+    "    let resultArray = ForEach(MResetZ, queryRegister);\n",
+    "\n",
+    "    return All(IsResultZero, resultArray);\n",
+    "}\n",
+    "\n",
+    "operation DeutschJozsa (n : Int) : Bool {\n",
+    "    using ((queryRegister, target) = (Qubit[n], Qubit())) {\n",
+    "        let res = IsConstantBooleanFunction(queryRegister, target, n);\n",
+    "        Reset(target);\n",
+    "        return res;\n",
+    "    }\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "outputs": [
+    {
+     "data": {
+      "application/json": "{\"Html\":\"<div id='execution-path-container-a6ca14ac-5bf8-4781-bf04-ac94436a1eea' />\"}",
+      "text/html": [
+       "<div id='execution-path-container-a6ca14ac-5bf8-4781-bf04-ac94436a1eea' />"
+      ],
+      "text/plain": [
+       "Microsoft.Quantum.IQSharp.Jupyter.DisplayableHtmlElement"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "%trace DeutschJozsa n=3"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "We passed in an extra argument `n=3` to the `%trace` command. Any argument that needs to be passed in to the given Q# operation is denoted by a key-value pair `<param>=<value>`. More details can be found in the documentation by running `%trace?`.\n",
+    "\n",
+    "Note that the visualization renders `IsConstantBooleanFunction` as a giant gate rather than its components. This is because, by default, the depth at which operations are traced out is at depth `1`.  To see the internals of `IsConstantBooleanFunction`, we need to render operations up to 2 levels deep by passing in the `--depth=2` flag. Try changing the value of the depth and see how the visualization changes!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/json": "{\"Html\":\"<div id='execution-path-container-fd9ca2ec-d3fc-4d28-8730-4f6134f916a8' />\"}",
+      "text/html": [
+       "<div id='execution-path-container-fd9ca2ec-d3fc-4d28-8730-4f6134f916a8' />"
+      ],
+      "text/plain": [
+       "Microsoft.Quantum.IQSharp.Jupyter.DisplayableHtmlElement"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "%trace DeutschJozsa n=3 --depth=2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Limitations of a quantum circuit\n",
```

Comment:
Ah, I see you have some of the difference between circuits and traces down here, that's great! It may make sense with the above, then, to point to this discussion.